### DISCUSSION TITLE
Markdown preview like Slate.js/Discord

### DISCUSSION THREAD

**Discussion by rxsto** [ueberdosis/tiptap#3004] [General Questions]
Hey there,

I'm currently trying to figure out how to replicate the "markdown" preview that Discord offers in their chat input field and message rendering 1:1.

I would experiment with Slate.js, but unfortunately it is only available for React, and I prefer creating web apps with Vue.

There has been previous discussion on this, see: https://github.com/ueberdosis/tiptap/issues/545 - but since this issue is more than 2 years old already, and since tiptap v2 is in public beta I wanted to check if there has been updates that might make this process easier.

So what I'd like to achieve is an implementation that handles various specific formats, like italics and bold (possibly via regex) and is able to apply custom styles without removing the actual decorators (and styling them differently).

Example:
![image](https://user-images.githubusercontent.com/30745852/179968330-70f92bb4-f155-4616-9477-d0f690c3058d.png)

I've read about Nodes and Marks, and tried to implement a similar behavior, but I haven't succeeded in any way yet. I'd be very grateful for any help and tips that might lead me into the right direction. I've tried various other editor frameworks and tiptap has looked the most promising as of now.

Cheers,
rxsto

**Comment by joakim** [comment]
I'm also trying to achieve this. I think it's the best way to edit Markdown, and I think TipTap should be a great editor for this. There's no need to automagically hide markup symbols, they would just activate commands. The text you'd get out would then be regular Markdown.

I'm sure it's not as easy as it sounds. You'd have to manage pasting differently. And validation of Markdown could be tricky (it has its intricacies).
